code/obj/machinery/door/door_control.dm

```diff
@@ -478,6 +490,47 @@ ADMIN_INTERACT_PROCS(/obj/machinery/door_control, proc/toggle)
 		playsound(src.loc, 'sound/impact_sounds/Generic_Click_1.ogg', 50, 1)
 	else
 		boutput(user, "<span class='alert'>It's broken.</span>")
+// Stolen from the vending module
+/// For a flying chat and message addition upon controller activation, not called outside of a child as things stand
+/obj/machinery/door_control/proc/speak(var/message)
+	ON_COOLDOWN(src, "scanner_speak", 2)
+	if ((src.status & NOPOWER) || !message)
+		return
+	else
+		welcome_text = make_chat_maptext(src, message, "color: [src.welcome_text_color];", alpha = src.welcome_text_alpha)
+		src.audible_message("<span class='subtle'><span class='game say'><span class='name'>[src]</span> beeps, \"[message]\"</span></span>", 2, assoc_maptext = welcome_text)
+		if (welcome_text && src.chat_text && length(src.chat_text.lines))
+			welcome_text.measure(src)
+			ON_COOLDOWN(src, "scanner_speak", 2)
+			for (var/image/chat_maptext/I in src.chat_text.lines)
+				if (I != welcome_text)
+					I.bump_up(welcome_text.measured_height)
+					ON_COOLDOWN(src, "scanner_speak", 2)
+		ON_COOLDOWN(src, "scanner_speak", 2)
+	return
+/// for sleepers entering listening post
+/obj/machinery/door_control/antagscanner
+	/// For the front door having a flying chat message or not.
+	var/entrance_scanner = 0
+	name = "Dubious Hand Scanner"
+	id = "Sleeper_Access"
+	icon = 'icons/obj/decoration.dmi'
+	icon_state = "antagscanner"
+	unpressed_icon = "antagscanner"
+	pressed_icon = "antagscanner-u"
+	unpowered_icon = "antagscanner" // should never happen, this is a failsafe if anything.
+	requires_power = 0
+	welcome_text = "Welcome, Agent."
+
+/obj/machinery/door_control/antagscanner/attack_hand(mob/user)
+	playsound(src.loc, 'sound/effects/handscan.ogg', 50, 1)
```

Comment:
```suggestion
	if (ON_COOLDOWN(src, "scan", 2 SECONDS))
		return
	playsound(src.loc, 'sound/effects/handscan.ogg', 50, 1)
```
not sure what the other `ON_COOLDOWN` calls are trying to do, but this is how you'd add a cooldown to the scanner